File: DoctorAPP/shared/api-contract.ts
// shared/api-contract.ts
// Canonical API types — backend Python mirrors these exactly.

export interface MetricDataPoint {
  date: string;
  value: number;
  unit: string;
  flag?: string;
}

export interface LongitudinalDataPoint {
  week_start: string;
  value: number;
  unit: string;
  trend?: string;
}

export interface AcuteMetrics {
  heartRateVariabilitySDNN: MetricDataPoint[];
  restingHeartRate: MetricDataPoint[];
  appleSleepingWristTemperature: MetricDataPoint[];
  respiratoryRate: MetricDataPoint[];
  walkingAsymmetryPercentage: MetricDataPoint[];
  stepCount: MetricDataPoint[];
  sleepAnalysis_awakeSegments: MetricDataPoint[];
}

export interface LongitudinalMetrics {
  restingHeartRate: LongitudinalDataPoint[];
  walkingAsymmetryPercentage: LongitudinalDataPoint[];
}

export interface PatientPayload {
  patient_id: string;
  sync_timestamp: string;
  hardware_source: string;
  patient_narrative: string;
  data: {
    acute_7_day: {
      granularity: string;
      metrics: AcuteMetrics;
    };
    longitudinal_6_month: {
      granularity: string;
      metrics: LongitudinalMetrics;
    };
  };
}

export interface ClinicalBrief {
  summary: string;
  key_symptoms: string[];
  severity_assessment: string;
  recommended_actions: string[];
}

export interface BiometricDelta {
  metric: string;
  acute_avg: number;
  longitudinal_avg: number;
  delta: number;
  unit: string;
  clinically_significant: boolean;
}

export interface ConditionMatch {
  condition: string;
  similarity_score: number;
  pmcid: string;
  title: string;
  snippet: string;
}

export interface AnalysisResponse {
  patient_id: string;
  clinical_brief: ClinicalBrief;
  biometric_deltas: BiometricDelta[];
  condition_matches: ConditionMatch[];
}

File: DoctorAPP/shared/api_contract.py
"""Canonical API types — frontend TypeScript mirrors these exactly."""

from pydantic import BaseModel
from typing import Union


class MetricDataPoint(BaseModel):
    date: str
    value: float
    unit: str
    flag: Union[str, None] = None


class LongitudinalDataPoint(BaseModel):
    week_start: str
    value: float
    unit: str
    trend: Union[str, None] = None


class AcuteMetrics(BaseModel):
    heartRateVariabilitySDNN: list[MetricDataPoint]
    restingHeartRate: list[MetricDataPoint]
    appleSleepingWristTemperature: list[MetricDataPoint]
    respiratoryRate: list[MetricDataPoint]
    walkingAsymmetryPercentage: list[MetricDataPoint]
    stepCount: list[MetricDataPoint]
    sleepAnalysis_awakeSegments: list[MetricDataPoint]


class LongitudinalMetrics(BaseModel):
    restingHeartRate: list[LongitudinalDataPoint]
    walkingAsymmetryPercentage: list[LongitudinalDataPoint]


class AcuteData(BaseModel):
    granularity: str
    metrics: AcuteMetrics


class LongitudinalData(BaseModel):
    granularity: str
    metrics: LongitudinalMetrics


class PatientData(BaseModel):
    acute_7_day: AcuteData
    longitudinal_6_month: LongitudinalData


class PatientPayload(BaseModel):
    patient_id: str
    sync_timestamp: str
    hardware_source: str
    patient_narrative: str
    data: PatientData


class ClinicalBrief(BaseModel):
    summary: str
    key_symptoms: list[str]
    severity_assessment: str
    recommended_actions: list[str]


class BiometricDelta(BaseModel):
    metric: str
    acute_avg: float
    longitudinal_avg: float
    delta: float
    unit: str
    clinically_significant: bool


class ConditionMatch(BaseModel):
    condition: str
    similarity_score: float
    pmcid: str
    title: str
    snippet: str


class AnalysisResponse(BaseModel):
    patient_id: str
    clinical_brief: ClinicalBrief
    biometric_deltas: list[BiometricDelta]
    condition_matches: list[ConditionMatch]
